After switching the Admin UI theme to "Base," an error occurred when attempting to access the keycloak login

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

admin/ui

### Describe the bug

Hi,

We were able to install the quarkus based keycloak ( 21.1.1 ) and access its GUI. 
Later we changed the **Realm setting --> Themes --> Admin UI theme** to "base" and save it. 
Now after logging out from the UI if we again try to login to the keycloak UI we are facing the following error 

![image](https://github.com/keycloak/keycloak/assets/97670904/f8d3d68d-a5e4-4a6f-8323-f31c13c1b814)


### Version

21.1.1

### Expected behavior

After changing the Admin UI theme to "base" and signing out we should be directed to the keycloak UI page.

### Actual behavior

After changing the Admin UI theme to "base" and signing out we are not able to get the keycloak login page.

### How to Reproduce?

Described in detail in the above section

### Anything else?

_No response_

I am able to reproduce this and getting an exception with the following log:

<details>
  <summary>Show log</summary>
  
  ```log
2023-07-04 12:33:58,003 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-6) Uncaught server error: org.keycloak.theme.FreeMarkerException: Failed to process template index.ftl
	at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.processTemplate(DefaultFreeMarkerProvider.java:52)
	at org.keycloak.services.resources.admin.AdminConsole.getMainPage(AdminConsole.java:354)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:154)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:118)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:560)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:452)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:413)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:415)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:378)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:174)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:131)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:33)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:429)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:240)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:154)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:157)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:229)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:82)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:147)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:84)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:44)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:58)
	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:36)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
	at org.keycloak.quarkus.runtime.integration.web.QuarkusRequestFilter.lambda$createBlockingHandler$0(QuarkusRequestFilter.java:82)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: freemarker.template.TemplateNotFoundException: Template not found for name "index.ftl".
The name was interpreted by this TemplateLoader: org.keycloak.theme.freemarker.DefaultFreeMarkerProvider$ThemeTemplateLoader@307efe8b.
	at freemarker.template.Configuration.getTemplate(Configuration.java:2883)
	at freemarker.template.Configuration.getTemplate(Configuration.java:2703)
	at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.getTemplate(DefaultFreeMarkerProvider.java:66)
	at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.processTemplate(DefaultFreeMarkerProvider.java:45)
	... 42 more
  ```
</details>

Looks like the `index.ftl` template does not exist in the base theme. I don't quite understand why the base theme is even selectable, as far as I know these themes serve to be extended, and should not be set directly.

We're likely to get rid of the base themes in the future entirely, so I am not quite sure how we want to handle this bug.

--- a/js/apps/admin-ui/cypress/e2e/realm_settings_tabs_test.spec.ts
+++ b/js/apps/admin-ui/cypress/e2e/realm_settings_tabs_test.spec.ts
@@ -126,7 +126,6 @@ describe("Realm settings tabs tests", () => {
 
     realmSettingsPage.selectLoginThemeType("keycloak");
     realmSettingsPage.selectAccountThemeType("keycloak");
-    realmSettingsPage.selectAdminThemeType("base");
     realmSettingsPage.selectEmailThemeType("base");
 
     realmSettingsPage.saveThemes();

--- a/js/apps/admin-ui/src/realm-settings/ThemesTab.tsx
+++ b/js/apps/admin-ui/src/realm-settings/ThemesTab.tsx
@@ -120,15 +120,17 @@ export const RealmSettingsThemesTab = ({
                 placeholderText="Select a theme"
                 data-testid="select-account-theme"
               >
-                {themeTypes.account.map((theme, idx) => (
-                  <SelectOption
-                    selected={theme.name === field.value}
-                    key={`account-theme-${idx}`}
-                    value={theme.name}
-                  >
-                    {t(`${theme.name}`)}
-                  </SelectOption>
-                ))}
+                {themeTypes.account
+                  .filter((theme) => theme.name !== "base")
+                  .map((theme, idx) => (
+                    <SelectOption
+                      selected={theme.name === field.value}
+                      key={`account-theme-${idx}`}
+                      value={theme.name}
+                    >
+                      {t(`${theme.name}`)}
+                    </SelectOption>
+                  ))}
               </Select>
             )}
           />
@@ -162,15 +164,17 @@ export const RealmSettingsThemesTab = ({
                 placeholderText="Select a theme"
                 data-testid="select-admin-theme"
               >
-                {themeTypes.admin.map((theme, idx) => (
-                  <SelectOption
-                    selected={theme.name === field.value}
-                    key={`admin-theme-${idx}`}
-                    value={theme.name}
-                  >
-                    {t(`${theme.name}`)}
-                  </SelectOption>
-                ))}
+                {themeTypes.admin
+                  .filter((theme) => theme.name !== "base")
+                  .map((theme, idx) => (
+                    <SelectOption
+                      selected={theme.name === field.value}
+                      key={`admin-theme-${idx}`}
+                      value={theme.name}
+                    >
+                      {t(`${theme.name}`)}
+                    </SelectOption>
+                  ))}
               </Select>
             )}
           />

